src/pip/_internal/req/req_install.py

```diff
@@ -769,8 +769,10 @@ def install(self, install_options, global_options=None, root=None,
         # Options specified in requirements file override those
         # specified on the command line, since the last option given
         # to setup.py is the one that is used.
-        global_options += self.options.get('global_options', [])
-        install_options += self.options.get('install_options', [])
+        global_options = \
+            global_options + self.options.get('global_options', [])
+        install_options = \
+            install_options + self.options.get('install_options', [])
```

Comment:
It's not at all clear why this makes a difference (I'm assuming it's something to do with in-place modification vs creating a new list?) so at a minimum, I'd want a comment explaining what's going on here. Better still would be a code change that didn't result in the reader having a huge temptation to switch back to the `+=` form 😄

Reply:
Reading the conversation thread clarifies (must read things in the right order in future :-)) but I'd still like a comment in the code.

Maybe better would be

```python
# Take a copy so we don't mutate the value supplied by the caller
global_options = list(global_options)
global_options += self.options.get('global_options', [])
```

as that makes the point explicit in the code as well as in the comment.

Reply:
+1

Having a comment would definitely be helpful here. Nice catch btw.

Reply:
Hello @simnalamburt! Thanks for spotting this. 😄 

If I may pitch in... If we are to take a copy, I would suggest to put it in another variable.
Currently, the `global_options` and `install_options` first refer to the arguments of the method, then refer to a modified copy of them. It can get confusing.

I would suggest something along:
```python
        package_global_options = \
            global_options + self.options.get('global_options', [])
        package_install_options = \
            install_options + self.options.get('install_options', [])
```

Of course renaming in the next usages as well would be required.

I don't know if the comment should be put in that case, but I believe it is at least much less tempting to refactor into the `+=` form with different names.

Reply:
> If I may pitch in... If we are to take a copy, I would suggest to put it in another variable.

It seems to be the best of the ideas so far. I'll update the PR.